From f4dd5366ff393eccec24eaba7c1acb78ac83197c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Tue, 20 Aug 2024 20:29:53 +0800
Subject: [PATCH 202/360] pinctrl: zhaoxin: fix build on >= 6.7

pinctrl_gpio_direction_{in,out}put() was revised post-6.6 to include a second
parameter, with the first being the chip and second being the offset as
received by the pinctrl I/O.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 drivers/pinctrl/zhaoxin/pinctrl-zhaoxin.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/zhaoxin/pinctrl-zhaoxin.c b/drivers/pinctrl/zhaoxin/pinctrl-zhaoxin.c
index bfd1bb7f022b..b57aa2d1f7a8 100644
--- a/drivers/pinctrl/zhaoxin/pinctrl-zhaoxin.c
+++ b/drivers/pinctrl/zhaoxin/pinctrl-zhaoxin.c
@@ -272,12 +272,12 @@ static void zhaoxin_gpio_set(struct gpio_chip *chip, unsigned int offset, int va
 
 static int zhaoxin_gpio_direction_input(struct gpio_chip *chip, unsigned int offset)
 {
-	return pinctrl_gpio_direction_input(chip->base + offset);
+	return pinctrl_gpio_direction_input(chip, offset);
 }
 
 static int zhaoxin_gpio_direction_output(struct gpio_chip *chip, unsigned int offset, int value)
 {
-	return pinctrl_gpio_direction_output(chip->base + offset);
+	return pinctrl_gpio_direction_output(chip, offset);
 }
 
 static int zhaoxin_gpio_request(struct gpio_chip *gc, unsigned int offset)
-- 
2.20.1

